### PR TITLE
Created a License header Checker

### PR DESCRIPTION
**Pull request by Divyaasm**
### Description
-Created a workflow to to check for Header Licenses in files.
-Added license Header for files missing them. 
-Updated files include - " python,groovy,js,ts,sh "

### Issues Resolved
Workflow checks for the presence of Header License on push and pull requests.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: vars/archiveAssembleUpload.groovy [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+* Copyright OpenSearch Contributors
```
Comment (gaiksaya) [inline vars/archiveAssembleUpload.groovy:2]: Can you fix the indentation for all groovy files?
Thanks!

**Review by gaiksaya — CHANGES_REQUESTED** [review]

**Comment by prudhvigodithi** [comment]
Hey @Divyaasm qq, will the License header Checker, check's the entire following lines? 
```
/*
* Copyright OpenSearch Contributors  
* SPDX-License-Identifier: Apache-2.0
* 
* The OpenSearch Contributors require contributions made to
* this file be licensed under the Apache-2.0 license or a
* compatible open source license.
*/
```

or just `Copyright OpenSearch Contributors ` and `SPDX-License-Identifier: Apache-2.0` based on the file `.licenserc.json`

File: vars/archiveAssembleUpload.groovy (hunk 1 above)
Comment (Divyaasm) [reply vars/archiveAssembleUpload.groovy:2]: Sure Sayali.

**Comment by Divyaasm** [comment]
> Hey @Divyaasm qq, will the License header Checker, check's the entire following lines?
> 
> ```
> /*
> * Copyright OpenSearch Contributors  
> * SPDX-License-Identifier: Apache-2.0
> * 
> * The OpenSearch Contributors require contributions made to
> * this file be licensed under the Apache-2.0 license or a
> * compatible open source license.
> */
> ```
> 
> or just `Copyright OpenSearch Contributors ` and `SPDX-License-Identifier: Apache-2.0` based on the file `.licenserc.json`

It checks for the two lines `Copyright OpenSearch Contributors ` and `SPDX-License-Identifier: Apache-2.0`. Barani recommended that these are the lines that are necessary. So, I've opted them.

**Comment by prudhvigodithi** [comment]
> It checks for the two lines `Copyright OpenSearch Contributors ` and `SPDX-License-Identifier: Apache-2.0`. Barani recommended that these are the lines that are necessary. So, I've opted them.

Got it thanks @Divyaasm , but I believe the entire sentence is a license
```
Copyright OpenSearch Contributors  
SPDX-License-Identifier: Apache-2.0

The OpenSearch Contributors require contributions made to
this file be licensed under the Apache-2.0 license or a
compatible open source license.
```
From Core https://github.com/opensearch-project/OpenSearch/blob/main/buildSrc/src/main/java/org/opensearch/gradle/pluginzip/Publish.java#L1-L7
@gaiksaya @peterzhuamazon @bbarani

**Comment by bbarani** [comment]
> Copyright OpenSearch Contributors  
> SPDX-License-Identifier: Apache-2.0

@prudhvigodithi We ran a campaign and used the 2 liner during that time as mentioned [here](https://github.com/opensearch-project/alerting/issues/136#issuecomment-966657180). @dblock Do you have any feedback on this license text?

**Review by gaiksaya — APPROVED** [review]
LGTM. Will leave to @bbarani and @dblock to merge this if they are okay.

**Pull request merged**